test/releasers/java-backport.ts

```diff
@@ -0,0 +1,630 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it, afterEach, beforeEach} from 'mocha';
+import * as nock from 'nock';
+nock.disableNetConnect();
+
+import {JavaBackport} from '../../src/releasers/java-backport';
+import * as suggester from 'code-suggester';
+import * as sinon from 'sinon';
+import {GitHubFileContents, GitHub} from '../../src/github';
+import {expect} from 'chai';
+import {buildGitHubFileContent} from './utils';
+import {buildMockCommit, stubSuggesterWithSnapshot} from '../helpers';
+import {readFileSync} from 'fs';
+import {resolve} from 'path';
+import {ReleasePR} from '../../src/release-pr';
+
+const sandbox = sinon.createSandbox();
+
+function buildFileContent(fixture: string): GitHubFileContents {
+  return buildGitHubFileContent(
+    './test/releasers/fixtures/java-yoshi',
+    fixture
+  );
+}
+
+describe('JavaBackport', () => {
+  afterEach(() => {
+    sandbox.restore();
+  });
+  it('creates a release PR', async function () {
+    const releasePR = new JavaBackport({
+      github: new GitHub({owner: 'googleapis', repo: 'java-trace'}),
+      packageName: 'java-trace',
+    });
+
+    sandbox
+      .stub(releasePR.gh, 'getRepositoryDefaultBranch')
+      .returns(Promise.resolve('master'));
+
+    // No open release PRs, so create a new release PR
+    sandbox
+      .stub(releasePR.gh, 'findOpenReleasePRs')
+      .returns(Promise.resolve([]));
+
+    // Indicates that there are no PRs currently waiting to be released:
+    sandbox
+      .stub(releasePR.gh, 'findMergedReleasePR')
+      .returns(Promise.resolve(undefined));
+
+    sandbox.stub(releasePR, 'latestTag').returns(
+      Promise.resolve({
+        name: 'v0.20.3',
+        sha: 'abc123',
+        version: '0.20.3',
```

Comment:
Can I expect an assertion in which this "0.20.3" becomes "0.20.4" after applying JavaBackport strategy? (If this is not the point of this test case, then never mind)

Reply:
For these tests, it's captured in the snapshot file (which records the data sent to the api calls).